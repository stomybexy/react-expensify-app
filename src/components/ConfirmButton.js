import React, { Component } from 'react';
import ConfirmModal from './ConfirmModal';

export default class ConfirmButton extends Component {

    state = {
        modalOpen: false
    };

    requestConfirm = () => this.setState({ modalOpen: true });

    onCancel = () => this.setState({ modalOpen: false });

    render() {
        return (
            <div>
                <button
                    className={this.props.className}
                    onClick={this.requestConfirm}>
                    {this.props.children}
                </button>

                <ConfirmModal
                    title={this.props.confirmTitle}
                    message={this.props.confirmMessage}
                    isOpen={this.state.modalOpen}
                    onConfirm={this.props.onConfirm}
                    onCancel={this.onCancel}
                />
            </div>
        );

    }
}
