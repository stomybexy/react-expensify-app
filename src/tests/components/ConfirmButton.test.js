import React from 'react';
import { shallow } from 'enzyme';
import ConfirmButton from '../../components/ConfirmButton';

let buttonText, onConfirm, wrapper;

beforeEach(() => {
    onConfirm = jest.fn();
    buttonText = 'Remove';
    wrapper = shallow(
        <ConfirmButton
            onConfirm={onConfirm}
            className="button button--secondary"
            confirmTitle="Confirm removal"
            confirmMessage="Remove this item ?"
        >
            {buttonText}
        </ConfirmButton>
    )
});

test('should render ConfirmButton correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('It should request confirm with modal on button click', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.state().modalOpen).toBe(true);
});

test('It should close the modal on cancel and not execute action', () => {
    wrapper.find('ConfirmModal').props().onCancel();
    expect(wrapper.state().modalOpen).toBe(false);
    expect(onConfirm).toHaveBeenCalledTimes(0);
});

test('It should close the modal and execute action on confirm', () => {
    wrapper.find('ConfirmModal').props().onConfirm();
    expect(wrapper.state().modalOpen).toBe(false);
    expect(onConfirm).toHaveBeenCalledTimes(1);
});