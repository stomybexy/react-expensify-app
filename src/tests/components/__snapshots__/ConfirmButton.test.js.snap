// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmButton correctly 1`] = `
<div>
  <button
    className="button button--secondary"
    onClick={[Function]}
  >
    Remove
  </button>
  <ConfirmModal
    isOpen={false}
    message="Remove this item ?"
    onCancel={[Function]}
    onConfirm={[Function]}
    title="Confirm removal"
  />
</div>
`;
